# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/nsPresContext.cpp
# Commit: f8ca02252dfc
# Full Hash: f8ca02252dfc48d7342703e6a635d510b9dc0b5d
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2020-04-02 09:51:45
# Regressor Bug: 1619349
# File Overlap Count: 1
# Description:
#   Bug 1619349 - patch 1 - Record statistics about font-matching behavior in the presContext. r=jwatt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D67787
# ==============================================================================

diff -r 01a2a527aa1a -r f8ca02252dfc layout/base/nsPresContext.cpp
--- a/layout/base/nsPresContext.cpp	Wed Apr 01 21:42:11 2020 +0000
+++ b/layout/base/nsPresContext.cpp	Wed Apr 01 21:42:36 2020 +0000
@@ -246,6 +246,11 @@
   if (Preferences::GetBool(GFX_MISSING_FONTS_NOTIFY_PREF)) {
     mMissingFonts = MakeUnique<gfxMissingFontRecorder>();
   }
+
+  // TODO: check whether this is too expensive to do for all pages;
+  // maybe we could throttle and only collect stats on 1% of page-loads,
+  // or something like that.
+  mFontStats = MakeUnique<FontMatchingStats>();
 }
 
 static const char* gExactCallbackPrefs[] = {